app/scripts/controllers/edit/buildConfig.js

```diff
@@ -117,6 +118,76 @@ angular.module('openshiftConsole')
       "SerialLatestOnly"
     ];
 
+    $scope.buildHookTypes = [{
+      "id": "script",
+      "label": "Shell Script"
+    }, {
+      "id": "command",
+      "label": "Command"
+    }, {
+      "id": "args",
+      "label": "Arguments to Default Image Entry Point"
+    }, {
+      "id": "scriptArgs",
+      "label": "Shell Script with Arguments"
+    }, {
+      "id": "commandArgs",
+      "label": "Command with Arguments"
+    }];
+
+    $scope.buildHookSelection = {
+      "type": {}
+    };
+
+    var getInitialBuildHookSelection = function() {
+      $scope.view.hasHooks = (_.size($scope.buildConfig.spec.postCommit.args) ||
+                              _.size($scope.buildConfig.spec.postCommit.command) ||
+                              $scope.buildConfig.spec.postCommit.script) ? true : false;
```

Comment:
`postCommit` is not guaranteed to be in the JSON, so need to handle that otherwise you could have a runtime error. Using `_.get` solves that.

```js
var hasArgs = !_.isEmpty(_.get($scope, 'buildConfig.spec.postCommit.args'));
var hasCommand = !_.isEmpty(_.get($scope, 'buildConfig.spec.postCommit.command'));
var hasScript = !!_.get($scope, 'buildConfig.spec.postCommit.command');
$scope.view.hasHooks = hasArgs || hasCommand || hasScript;
```

Then you can use the local vars below without checking size again.